indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -989,9 +990,13 @@ private void processDimensionsSpec(final QueryableIndex index)
           // column for it.
           final ColumnHolder columnHolder = index.getColumnHolder(dimension);
           if (columnHolder != null) {
+            DimensionSchema schema = columnHolder.getColumnFormat().getColumnSchema(dimension);
+            if (schema instanceof StringDimensionSchema) {
```

Comment:
A comment about why this is happening would be useful. It probably won't be obvious to a reader.